### PR TITLE
Feat - dubbo http proxy and dubbo triple proxy

### PR DESCRIPTION
**Pull request by ztelur**
**What this PR does**:

#### request protocol transform
- dubbo to http
- dubbo to triple
- triple to dubbo

samples are `samples/dubbohttpproxy` and `samples/dubbotripleproxy`

#### tcp listener

use getty to establish tcp server. refer to `pkg/listener/tcp/tcp_listener.go`

#### triple listener

use triple to establish triple or grpc server. refer to `pkg/listener/triple/triple_listener.go`

#### dubbo filter mechanism
refer to `pkg/common/extension/filter/filter.go` like http filter

### CODE REVIEW

File: pkg/context/dubbo/context.go [hunk 1]
```diff
@@ -0,0 +1,44 @@
+package dubbo
```
Comment (AlexStocks) [inline pkg/context/dubbo/context.go:18]: where is the license?

File: pkg/common/tcp/manager.go [hunk 2]
```diff
@@ -0,0 +1 @@
+package tcp
```
Comment (AlexStocks) [inline pkg/common/tcp/manager.go]: where is the license?

File: pkg/filterchain/network_filter_chain.go [hunk 3]
```diff
@@ -34,51 +38,74 @@ type NetworkFilterChain struct {
...
+		}
+
+		config := p.Config()
+		if err := yaml.ParseConfig(config, f.Config); err != nil {
+			logger.Error("CreateNetworkFilterChain %s parse config error %s", f.Name, err)
+		}
+
+		filter, err := p.CreateFilter(config, bs)
+		if err != nil {
+			logger.Error("CreateNetworkFilterChain %s createFilter error %s", f.Name, err)
```
Comment (mark4z) [inline pkg/filterchain/network_filter_chain.go:104]: may be miss the 'continue' ?

File: pkg/filter/network/dubboproxy/manager.go [hunk 4]
```diff
@@ -0,0 +1,198 @@
...
+	"github.com/apache/dubbo-go-pixiu/pkg/common/constant"
+	"github.com/apache/dubbo-go-pixiu/pkg/common/extension/filter"
+	router2 "github.com/apache/dubbo-go-pixiu/pkg/common/router"
+	dubbo2 "github.com/apache/dubbo-go-pixiu/pkg/context/dubbo"
+	"github.com/apache/dubbo-go-pixiu/pkg/logger"
+	"github.com/apache/dubbo-go-pixiu/pkg/model"
+)
+
+// HttpConnectionManager network filter for http
+type DubboProxyConnectionManager struct {
```
Comment (mark4z) [inline pkg/filter/network/dubboproxy/manager.go:50]: // HttpConnectionManager

**Review by mark4z — APPROVED** [review]

File: pkg/filterchain/network_filter_chain.go (hunk 3 above)
Comment (ztelur) [reply pkg/filterchain/network_filter_chain.go:104]: Done

File: pkg/filter/network/dubboproxy/manager.go (hunk 4 above)
Comment (ztelur) [reply pkg/filter/network/dubboproxy/manager.go:50]: Fix

File: pkg/filter/network/dubboproxy/dubbofiltermanager.go [hunk 5]
```diff
@@ -0,0 +1,61 @@
...
+		}
+
+		config := p.Config()
+		if err := yaml.ParseConfig(config, f.Config); err != nil {
+			logger.Error("createDubboFilter %s parse config error %s", f.Name, err)
+		}
+
+		filter, err := p.CreateFilter(config)
+		if err != nil {
+			logger.Error("createDubboFilter %s createFilter error %s", f.Name, err)
```
Comment (mark4z) [inline pkg/filter/network/dubboproxy/dubbofiltermanager.go:58]: Miss 'continue'

**Review by mark4z — APPROVED** [review]

File: pkg/filter/network/dubboproxy/filter/http/httpfilter.go [hunk 6]
```diff
@@ -0,0 +1,151 @@
...
+)
+
+import (
+	"github.com/apache/dubbo-go-pixiu/pkg/common/constant"
+	"github.com/apache/dubbo-go-pixiu/pkg/common/extension/filter"
+	dubbo2 "github.com/apache/dubbo-go-pixiu/pkg/context/dubbo"
+	"github.com/apache/dubbo-go-pixiu/pkg/server"
+)
+
+const (
```
Comment (AlexStocks) [inline pkg/filter/network/dubboproxy/filter/http/httpfilter.go:44]: 常量都是大写。Go 变量名字里名字里不要加下划线。x_dubbo_http_dubbo_version -> XDubboHttpDubboVersion

**Review by AlexStocks — APPROVED** [review]

**Review by xiaoliu10 — APPROVED** [review]

**Review by PhilYue — APPROVED** [review]

**Pull request merged**